File: 1800-maximum-ascending-subarray-sum/1800-maximum-ascending-subarray-sum.java
class Solution {
    public int maxAscendingSum(int[] nums) {
        int n = nums.length;

        int sum = nums[0];
        int temp = nums[0];

        for(int i = 1; i < nums.length; i++) {
            if(nums[i] > nums[i - 1]) {
                temp += nums[i];
            } else {
                sum = Math.max(sum, temp);
                temp = nums[i];
            }
        }

        sum = Math.max(sum, temp);

        return sum;
    }
}

File: 1800-maximum-ascending-subarray-sum/1800-maximum-ascending-subarray-sum.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxAscendingSum = function (nums) {
    let currMax = nums[0];
    let actMax = nums[0];

    for (let i = 1; i < nums.length; i++) {
        let preVal = nums[i - 1];
        let val = nums[i];

        if (preVal < val) {
            currMax += val;
        } else {
            actMax = Math.max(currMax, actMax);

            currMax = val;
        }

        actMax = Math.max(currMax, actMax);
    }

    return actMax;
};